client/gutenberg/editor/edit-post/hooks/components/index.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+const replaceMediaUpload = () => null;
+
+addFilter(
```

Comment:
This is going to be the way to hook in Calypso's custom Media Gallery implementation.